R/calculateUnifrac.R

```diff
@@ -265,121 +265,8 @@ runUnifrac <- function(x, tree, weighted = FALSE, normalized = TRUE,
-                                               tree = tree,
-                                               samplesums = samplesums,
-                                               edge_array = edge_array,
-                                               BPPARAM = BPPARAM)
-        } else {
-            # This is only relevant to weighted-Unifrac.
-            # For denominator in the normalized distance, we need the age of each
-            # tip.
-            # 'z' is the tree in postorder order used in calls to .C
-            # Descending order of left-hand side of edge (the ancestor to the node)
-            z <- ape::reorder.phylo(tree, order="postorder")
-            # Call phyloseq-internal function that in-turn calls ape's internal
-            # horizontal position function, in C, using the re-ordered phylo object,
-            # `z`
-            tipAges = node.depth.edgelength(tree)
-            # Keep only the tips, and add the tip labels in case `z` order differs
-            # from `tree`
-            tipAges <- tipAges[seq.int(1L, length(tree$tip.label))]
-            names(tipAges) <- z$tip.label
-            # Explicitly re-order tipAges to match x
-            tipAges <- tipAges[rownames(x)]
-            distlist <- BiocParallel::bplapply(spn,
-                                               unifrac_weighted_norm,
-                                               mat = x,
-                                               tree = tree,
-                                               samplesums = samplesums,
-                                               edge_array = edge_array,
-                                               tipAges = tipAges,
-                                               BPPARAM = BPPARAM)
-        }
-    } else {
-        # For unweighted Unifrac, convert the edge_array to an occurrence
-        # (presence/absence binary) array
-        edge_occ <- (edge_array > 0) - 0
-        distlist <- BiocParallel::bplapply(spn,
-                                           unifrac_unweighted,
-                                           tree = tree,
-                                           samplesums = samplesums,
-                                           edge_occ = edge_occ,
-                                           BPPARAM = BPPARAM)
-    }
-    # Initialize UnifracMat with NAs
-    UnifracMat <- matrix(NA_real_, ncol(x), ncol(x))
-    rownames(UnifracMat) <- colnames(UnifracMat) <- colnames(x)
-    # Matrix-assign lower-triangle of UnifracMat. Then coerce to dist and
-    # return.
-    matIndices <- matIndices <- matrix(c(vapply(spn,"[",character(1),2L),
-                                         vapply(spn,"[",character(1),1L)),
-                                       ncol = 2)
-    UnifracMat[matIndices] <- unlist(distlist)
-    #
-    stats::as.dist(UnifracMat)
+    .calculate_distance(x, FUN = rbiom::unifrac, tree = tree, 
+                        weighted = weighted)
```

Comment:
You can just call
`
unifrac(x, tree = tree, weighted = weighted)`

Add rbiom to @importFrom

Reply:
You can also remove those lines from the documentation that describes the older implementation

Reply:
And remove those functions from importFrom that are not used anymore